_episodes/03-sharing.md

```diff
@@ -95,6 +95,150 @@ origin  https://github.com/<your_github_username>/hello-world (push)
 ~~~
 {: .output}
 
+## SSH Background and Setup
+We still need to do a little more setup before we can actually connect to this remote repository.
+We need to set up a way for our local computer to authenticate with GitHub so that GitHub recognizes our computer as belonging to the same person who owns the GitHub repository. 
+
+We will use SSH as our authentication method. SSH stands for Secure Shell Protocol.  SSH is a cryptographic network protocol that allows secure communication between computers using an otherwise insecure network.
+
+SSH uses what is called a key pair -- two keys that work together to validate access. One key is used publicly (the public key) and the other key is kept private (the private key).
+
+You can think of the public key as a padlock, and only you have the key (the private key) to open it. You use the public key where you want a secure method of communication, such as your GitHub account.  You give this padlock, or public key, to GitHub and say “lock the communications to my account with this so that only computers that have my private key can unlock communications and send Git commands as my GitHub account.”
+
+What we will do now is the minimum required to set up the SSH keys and add the public key to a GitHub account.
+I'm not going to lie, this is a bit tedious and confusing. But you have to do it to get to the fun part, so hang in there.
+
+<!-- > ## Advanced SSH -->
+<!-- > A supplemental episode in this lesson discusses SSH and key pairs in more depth and detail.-->
+<!-- {: .callout}-->
+
+The first thing we are going to do is check if this has already been done on the computer you’re on.  Because generally speaking, this setup only needs to happen once and then you can forget about it.
+
+> ## Keeping your keys secure
+> You shouldn't really forget about your SSH keys, since they keep your account secure. It’s good
+>  practice to audit your secure shell keys every so often. Especially if you are using multiple
+>  computers to access your account.
+{: .callout}
+
+We will run the list command (`ls`) to check what key pairs already exist on your computer. 
+In our command we use the `~` as the shorthand for "my home directory."
+
+~~~
+ls -al ~/.ssh
+~~~
+{: .language-bash}
+
+Your output is going to look a little different depending on whether or not SSH has ever been set up on the computer you are using.
```

Comment:
from @doingarchives's comment:

> Your output is going (131) - this information is not really adding anything instructional (it is observational and individual experiences will vary confusingly)

I think this part of the lesson is for the instructor to figure out who actually needs to create a key, and who already has one and can skip the next few commands on creating a key.
We can make this distinction more explicit here and maybe say something like "if you do have existing keys, skip the next section and go here: xxxx" and link to the part where everyone copies their `.pub` file to paste into GitHub.

Reply:
@chendaniely In my instructor experience - the availability of a user account is a set up issue ... the key should already be in the possession of the person taking the class. Often this is the case in most types of software training - I recognise @emcaulay is wrestling with this issue as far as where it goes in the lesson plan ... the point of the lesson is not to 'set up' Git for a user / learner but to demonstrate the typical use case for a professional. The somewhat repugnant 'savvy librarian' like the characters on moralistic fiction would already come to class with their Git user credentials. To elevate the lesson I think we should address the credential issue (and any security needs) as a set up item, then merely reference the password in the lesson body. One could add a - 'Oops!? No password, see the set up instructions you completed prior to the course.

COVID-19 and the learner caps in place at most UC institutional offerings means that for every registered user wanting a live demo - there are many others who do the lesson remotely. As this trend accelerates, the need to maintain lessons that are 'self contained' and 'self referential' is a significant policy priority.

Socialising the discussion and perhaps blocking out times and best channels is a great first step for the moderator / maintainer cadre. At UC San Diego Slack was always open so it was easier to be timely in responding. At Harvard (alas) Teams is the norm - and that is almost nowhere to be found in the Carpentries ecoverse.

Reply:
I also have instructor experience and am a librarian, and my experience differs from yours @doingarchives. I think set up is an integral part of this lesson. GitHub and private and public keys are not part of our target demographic's general activities. The goal of this lesson and the core LC workshop series is to enable the learner to adopt these technologies and I believe the Git and GitHub set up is sufficiently alien that it should be in the lesson not outside the lesson. The much more sophisticated audience and extensive lesson in the Software Carpentry program also has decided to include the set up within the lesson. 

As a separate matter, I have removed the use of "Savvy Librarian" as you requested.

Reply:
As far as "socialising the discussion," I think you mean have a meeting? But I'm not certain since that phrase is unfamiliar to me. 

I think since you were on-boarded, @chendaniely has taken over the role of Maintainers Coordinator across Library, Data, and Software Carpentry. He hosts 2 meetings a month (with 2 different time slots to accommodate the wide range of time zones). The notes and connection information are here: https://codimd.carpentries.org/maintainers?both


I am still learning the different methods of communication for the Carpentries Maintainers, but there is definitely:

- TopicBox: https://carpentries.topicbox.com/groups/maintainers
- Slack: swcarpentry.slack.com(i recognize you are saying you do not regularly use that)

Reply:
I agree with @emcaulay here about the overall goals for the lesson including helping learners with setup complications. This is a core part of the "meet the learner where they are" principle from Instructor Training.